### DISCUSSION TITLE
Flag some environment variables as required by Node build container

### DISCUSSION THREAD

**Discussion by AugustMiller** [craftcms/cloud#33] [Feature Requests & Enhancements]
Projects may want to interact with privileged resources outside of the build container (like a third-party search API), but not bake credentials into the repository itself.

- [Cloud Build Process and Artifacts ](https://craftcms.com/knowledge-base/cloud-builds#node)
- The Digital Ocean App Platform allows you to do this, but [only via app spec](https://docs.digitalocean.com/products/app-platform/how-to/use-environment-variables/#define-build-time-environment-variables)

**Comment by timkelty** [comment]
It's worth noting that the node process builds what are presumably client-side assets, so anything actually "secret" wouldn't be appropriate to include.

**Reply by michtio** [reply]
I agree with you here, but in some build steps we need an API key to be read, e.g. in cases like SEARCH_ONLY_API from Typesense/Algolia and the works, for the CI/CD to actually pass, a connection needs to be established during the build process, otherwise it shouldn't be allowed to be deployed (responsibility of the `npm run build` script) but without actually having the possibility to run on FE variables it is tricky.

I support @brianjhanson suggestion beneath, this would actually be very nice, but it might become tricky if you want to keep your prefixed variables in the `.env` as most likely we'd need to hardcode it in the code either way. In many environments the front-end build artifacts could benefit of reading the `.env` file too.

**Comment by brianjhanson** [comment]
We could also an approach like Vite (and probably others) and automatically/only expose variables with a prefix to the node build.

**Reply by michtio** [reply]
@brianjhanson this is actually a great suggestion, this works like a charm with Vite. All in for this solution (This issue is actually one August wrote on my behalf ;))
